### PR TITLE
Swift migration

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
- Migrated to Swift 5.0, which requires Xcode 10.2
- Did a manual code hunt for implementations that could now be auto-synthesized
- Moved the CodingKeys enum to just after the struct/class definition 
- Updated projects to recommended settings

### CODE REVIEW

File: Research/Research.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3344,7 +3345,7 @@
 				DEFINES_MODULE = YES;
 				DEVELOPMENT_TEAM = KA9Z8R6M6K;
 				DYLIB_COMPATIBILITY_VERSION = 1;
-				DYLIB_CURRENT_VERSION = 30;
+				DYLIB_CURRENT_VERSION = 29;
```
Comment (Erin-Mounts) [inline Research/Research.xcodeproj/project.pbxproj]: hopefully not important and will get resolved with next stable release, but why is this going backward

File: Research/Research/RSDFactory.swift [hunk 2]
```diff
@@ -797,7 +797,7 @@ open class RSDFactory {
         case .file:
             return try RSDFileResultObject(from: decoder)
         default:
-            throw RSDValidationError.undefinedClassType("\(self) does not support `\(typeName)` as a decodable class type for a result.")
+            throw RSDValidationError.undefinedClassType("\(self) does not support `\(resultType)` as a decodable class type for a result.")
```
Comment (Erin-Mounts) [inline Research/Research/RSDFactory.swift:760]: parameters in doc comments above need to be updated, it looks like

File: ResearchUI/ResearchUI.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -703,7 +704,7 @@
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
-				CURRENT_PROJECT_VERSION = 30;
+				CURRENT_PROJECT_VERSION = 29;
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI.xcodeproj/project.pbxproj]: again, going backward? maybe not an issue though?

**Review by Erin-Mounts — APPROVED** [review]
Couple of nits/questions, otherwise looks pretty straightforward

File: Research/Research.xcodeproj/project.pbxproj (hunk 1 above)
Comment (syoung-smallwisdom) [reply Research/Research.xcodeproj/project.pbxproj]: huh

**Pull request merged**